### PR TITLE
[Draft] Manage ruy_context on runtime

### PR DESCRIPTION
**Pull request by periannath**
- Create ExternalContext class which has ruy context
  - CPU backend has ExternalContext
  - KernelGenerator and FullyConnectedLayer has ExternalContext
  - Pass ruy_context to cker::FullyConnectedHybrid
- Link ruy library to onert

ONE-DCO-1.0-Signed-off-by: JiHwan Yeo <jihwan.yeo@samsung.com>

---

For issue : #3318

### CODE REVIEW

**Comment by hseok-oh** [comment]
For more general solution
- Introduce `IExternalContext`

It `ruy_context` need for each executor, 
- Introduce `RuyContext` in cpu backend
- Register `RuyContext` object to generated executor object

**Comment by periannath** [comment]
@hseok-oh Thanks! I'll introduce `IExternalContext` as you commented. AFAIK, one `ruy_context` can be shared by executors.

**Comment by periannath** [comment]
@nnfw-bot test cross-runtime-test-release

File: runtime/onert/core/include/backend/IExternalContext.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace onert
+{
+namespace backend
+{
+
+struct IExternalContext
+{
+  virtual ~IExternalContext() = default;
+  virtual void SetMaxNumThreads(int) = 0;
```
Comment (wateret) [inline runtime/onert/core/include/backend/IExternalContext.h:28]: Method names must start with small "s"

**Comment by periannath** [comment]
#3559 is merged.

**Pull request closed without merging**